flake: TestKeyring "file does not exist"

```
=== Failed
=== FAIL: cli/sessionstore TestWindowsKeyring_WriteReadDelete (0.04s)
    sessionstore_windows_test.go:74: 
        	Error Trace:	C:/actions-runner/coder/coder/cli/sessionstore/sessionstore_windows_test.go:74
        	Error:      	Expected error with "file does not exist" in chain but got nil.
        	Test:       	TestWindowsKeyring_WriteReadDelete

=== FAIL: cli/sessionstore TestKeyring/StorageFormat (0.02s)
    sessionstore_test.go:251: 
        	Error Trace:	C:/actions-runner/coder/coder/cli/sessionstore/sessionstore_test.go:251
        	Error:      	Received unexpected error:
        	            	file does not exist
        	Test:       	TestKeyring/StorageFormat
    --- FAIL: TestKeyring/StorageFormat (0.02s)

=== FAIL: cli/sessionstore TestKeyring/MultipleServers (0.02s)
    sessionstore_test.go:179: 
        	Error Trace:	C:/actions-runner/coder/coder/cli/sessionstore/sessionstore_test.go:179
        	Error:      	Received unexpected error:
        	            	file does not exist
        	Test:       	TestKeyring/MultipleServers
    --- FAIL: TestKeyring/MultipleServers (0.02s)

=== FAIL: cli/sessionstore TestKeyring (0.00s)
```

seen here: https://github.com/coder/coder/runs/65100382395

--- a/cli/sessionstore/sessionstore_test.go
+++ b/cli/sessionstore/sessionstore_test.go
@@ -21,9 +21,8 @@ type storedCredentials map[string]struct {
 	APIToken string `json:"api_token"`
 }
 
+//nolint:paralleltest, tparallel // OS keyring is flaky under concurrent access
 func TestKeyring(t *testing.T) {
-	t.Parallel()
-
 	if runtime.GOOS != "windows" && runtime.GOOS != "darwin" {
 		t.Skip("linux is not supported yet")
 	}
@@ -37,8 +36,6 @@ func TestKeyring(t *testing.T) {
 	)
 
 	t.Run("ReadNonExistent", func(t *testing.T) {
-		t.Parallel()
-
 		backend := sessionstore.NewKeyringWithService(testhelpers.KeyringServiceName(t))
 		srvURL, err := url.Parse(testURL)
 		require.NoError(t, err)
@@ -50,8 +47,6 @@ func TestKeyring(t *testing.T) {
 	})
 
 	t.Run("DeleteNonExistent", func(t *testing.T) {
-		t.Parallel()
-
 		backend := sessionstore.NewKeyringWithService(testhelpers.KeyringServiceName(t))
 		srvURL, err := url.Parse(testURL)
 		require.NoError(t, err)
@@ -63,8 +58,6 @@ func TestKeyring(t *testing.T) {
 	})
 
 	t.Run("WriteAndRead", func(t *testing.T) {
-		t.Parallel()
-
 		backend := sessionstore.NewKeyringWithService(testhelpers.KeyringServiceName(t))
 		srvURL, err := url.Parse(testURL)
 		require.NoError(t, err)
@@ -91,8 +84,6 @@ func TestKeyring(t *testing.T) {
 	})
 
 	t.Run("WriteAndDelete", func(t *testing.T) {
-		t.Parallel()
-
 		backend := sessionstore.NewKeyringWithService(testhelpers.KeyringServiceName(t))
 		srvURL, err := url.Parse(testURL)
 		require.NoError(t, err)
@@ -115,8 +106,6 @@ func TestKeyring(t *testing.T) {
 	})
 
 	t.Run("OverwriteToken", func(t *testing.T) {
-		t.Parallel()
-
 		backend := sessionstore.NewKeyringWithService(testhelpers.KeyringServiceName(t))
 		srvURL, err := url.Parse(testURL)
 		require.NoError(t, err)
@@ -146,8 +135,6 @@ func TestKeyring(t *testing.T) {
 	})
 
 	t.Run("MultipleServers", func(t *testing.T) {
-		t.Parallel()
-
 		backend := sessionstore.NewKeyringWithService(testhelpers.KeyringServiceName(t))
 		srvURL, err := url.Parse(testURL)
 		require.NoError(t, err)
@@ -199,7 +186,6 @@ func TestKeyring(t *testing.T) {
 	})
 
 	t.Run("StorageFormat", func(t *testing.T) {
-		t.Parallel()
 		// The storage format must remain consistent to ensure we don't break
 		// compatibility with other Coder related applications that may read
 		// or decode the same credential.

--- a/cli/sessionstore/sessionstore_windows_test.go
+++ b/cli/sessionstore/sessionstore_windows_test.go
@@ -25,9 +25,8 @@ func readRawKeychainCredential(t *testing.T, serviceName string) []byte {
 	return winCred.CredentialBlob
 }
 
+//nolint:paralleltest, tparallel // OS keyring is flaky under concurrent access
 func TestWindowsKeyring_WriteReadDelete(t *testing.T) {
-	t.Parallel()
-
 	const testURL = "http://127.0.0.1:1337"
 	srvURL, err := url.Parse(testURL)
 	require.NoError(t, err)

